### PR TITLE
fix(nodebuilder/host): Ensure libp2p metrics are collected to prometheus

### PR DESCRIPTION
**Pull request by walldiss**
This PR fixes the bug of prometheus registry not being provided to libp2p constructor as well as few additional improvements of adjustment code:
- provide Registrer to libp2p if flag is enabled
- clarify telemetry enabling flag names and avoid overwrites

### CODE REVIEW

File: nodebuilder/p2p/metrics.go [hunk 1]
```diff
@@ -32,19 +33,27 @@ const (
...
 	labels := prometheus.Labels{
 		networkLabel:  network.String(),
 		nodeTypeLabel: nodeType.String(),
 		peerIDLabel:   peerID.String(),
 	}
 	wrapped := prometheus.WrapRegistererWith(labels, reg)
+	// Set the default global registerer to the wrapped one with labels. This way all the metrics
+	// registered with the default registerer will be labeled with the provided labels. It is important
+	// because unlike libp2p metrics, bitswap metrics are registered with the default global registerer.
+	prometheus.DefaultRegisterer = wrapped
```
Comment (Wondertan) [inline nodebuilder/p2p/metrics.go:56]: Aren't bitswap metrics registered on init causing this overwrite being unaffectual?

**Review by Wondertan — APPROVED** [review]

File: nodebuilder/p2p/metrics.go (hunk 1 above)
Comment (walldiss) [reply nodebuilder/p2p/metrics.go:56]: You are right, I need to ensure order of init:
1. update global
2. inject bitswap Registrer
3. start bitswap

**Review by Wondertan — DISMISSED** [review]

**Comment by Wondertan** [comment]
@walldiss, did you test it yet to see if it works with Graphana?

**Review by vgonkivs — APPROVED** [review]

**Comment by walldiss** [comment]
This PR is tested and proven to be working. Fixes related to bitswap metrics initialisation extracted to separate PR:
https://github.com/celestiaorg/celestia-node/pull/3763

**Review by renaynay — APPROVED** [review]
utack

**Pull request merged**